src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,110 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connectionutil.ConnectionUtil;
+import mate.academy.exeption.DataProcessingException;
+import mate.academy.model.Book;
+
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to create book", e);
```

Comment:
Consider providing more detailed error messages in the `DataProcessingException` to include the book details or the SQL query that failed. This can help with debugging and understanding the context of the error.